libcontainer/process_linux.go

```diff
@@ -340,6 +346,9 @@ func (p *setnsProcess) start() (retErr error) {
 	if err := p.comm.syncSockParent.Shutdown(unix.SHUT_WR); err != nil && ierr == nil {
 		return err
 	}
+	if !seenProcReady && ierr == nil {
+		ierr = errors.New("procReady not received")
+	}
```

Comment:
I think you don't need this bit.

Reply:
This just mirrors `runc create`. If the `runc init` process dies before sending `procReady` we can detect it here (`parseSync` just waits for `EOF`, it doesn't care about if the process died). AFAICS there are no other liveliness checks after `start()` returns.

Reply:
Makes sense.